baybe/objectives/base.py

```diff
@@ -0,0 +1,47 @@
+"""Base classes for all objectives."""
+
+from abc import ABC, abstractmethod
+from typing import Union
+
+import pandas as pd
+from attrs import define
+
+from baybe.objectives.deprecation import structure_objective
+from baybe.serialization.core import (
+    converter,
+    unstructure_base,
+)
+from baybe.serialization.mixin import SerialMixin
+from baybe.targets.base import Target
+
+
+@define(frozen=True)
+class Objective(ABC, SerialMixin):
+    """Abstract base class for all objectives."""
+
+    @property
+    @abstractmethod
+    def targets(self) -> tuple[Target, ...]:
+        """The targets included in the objective."""
+
+    @abstractmethod
+    def transform(self, data: pd.DataFrame) -> pd.DataFrame:
+        """Transform targets from experimental to computational representation.
+
+        Args:
+            data: The data to be transformed. Must contain columns for all targets
+                but can contain additional columns.
+
+        Returns:
+            A new dataframe with the targets in computational representation.
+        """
+
+
+def to_objective(x: Union[Target, Objective], /) -> Objective:
```

Comment:
is this needed for anythin else other than the converter? the functionality is so minimal Id prefer a lambda instead of something like this which needs to be imported from another file (from the perspective of the file where its used)

Reply:
Well, my thought was:
* we might use it also in other places where we pass objectives
* I'm actually not sure if the typing would work correctly without it because there still is that open `mypy` issue which required me to add the `to_tuple` function in the other PR. 

If you insist, I could test if it works without it. I saw that the new pydantic version now supports `typing.Annotated` which allows you to bind validation to types instead of attributes. This would be a nice alternative here, was already thinking about opening an `attrs` ticket and see if we could get something like that in the long run ...